### PR TITLE
zh-cn: update the translation of `URLSearchParams`

### PR DESCRIPTION
**Pull request by fuchunhui**
### Description
1. sync translate.

### CODE REVIEW

File: files/zh-cn/web/api/urlsearchparams/index.md [hunk 1]
```diff
@@ -53,46 +58,109 @@ for (const [key, value] of mySearchParams.entries()) {
...
 
-var paramsString2 = "?query=value";
-var searchParams2 = new URLSearchParams(paramsString2);
-searchParams2.has("query"); // true
+### 保留加号
 
-var url = new URL("http://example.com/search?query=%40");
-var searchParams3 = new URLSearchParams(url.search);
-searchParams3.has("query"); // true
+`URLSearchParams` 的构造函数将加号（`+`）解释为空格，这会有问题。在下面的示例中，我们使用[十六进制转义序列](/zh-CN/docs/Web/JavaScript/Reference/Lexical_grammar#hexadecimal_escape_sequences)模拟一个包含二进制数据（其中每个字节都携带信息）的字符串，该数据需要存储在 URL 搜索参数中。请注意 `btoa()` 生成的编码字符串包含 `+`，并不会被 `URLSearchParams` 保留。
```
Comment (yin1999) [inline files/zh-cn/web/api/urlsearchparams/index.md:129]: Always pay attention to the hashes

```suggestion
`URLSearchParams` 的构造函数将加号（`+`）解释为空格，这可能会有问题。在下面的示例中，我们使用[十六进制转义序列](/zh-CN/docs/Web/JavaScript/Reference/Lexical_grammar#十六进制转义序列)模拟一个包含二进制数据（其中每个字节都携带信息）的字符串，该数据需要存储在 URL 查询参数中。请注意 `btoa()` 生成的编码字符串包含 `+`，而其并不会被 `URLSearchParams` 保留。
```

File: files/zh-cn/web/api/urlsearchparams/index.md [hunk 2]
```diff
@@ -16,23 +16,28 @@ for (const [key, value] of mySearchParams.entries()) {
...
   - : 返回一个 `URLSearchParams` 对象。
 
-## 方法
+## 属性
 
-该接口不继承任何属性。
+- {{domxref("URLSearchParams.size", 'size')}} {{ReadOnlyInline}}
+  - : 返回 `URLSearchParams` 对象中搜索参数的总个数。
+
+## 方法
```
Comment (yin1999) [inline files/zh-cn/web/api/urlsearchparams/index.md:29]:
```suggestion
## 实例方法
```

File: files/zh-cn/web/api/urlsearchparams/index.md [hunk 3]
```diff
@@ -53,46 +58,109 @@ for (const [key, value] of mySearchParams.entries()) {
...
+const paramsString1 = "http://example.com/search?query=%40";
+const searchParams1 = new URLSearchParams(paramsString1);
+
+console.log(searchParams1.has("query")); // false
+console.log(searchParams1.has("http://example.com/search?query")); // true
 
-searchParams1.has("query"); // false
-searchParams1.has("http://example.com/search?query"); // true
+console.log(searchParams1.get("query")); // null
+console.log(searchParams1.get("http://example.com/search?query")); // "@" (equivalent to decodeURIComponent('%40'))
```
Comment (yin1999) [inline files/zh-cn/web/api/urlsearchparams/index.md:116]:
```suggestion
console.log(searchParams1.get("http://example.com/search?query")); // "@"（等价于 decodeURIComponent('%40')）
```

File: files/zh-cn/web/api/urlsearchparams/index.md (hunk 1 above)
Comment (fuchunhui) [reply files/zh-cn/web/api/urlsearchparams/index.md:129]: `查询参数` are more accurate, but I found that in other places, what was previously written was the `搜索参数`, so I kept it consistent.

File: files/zh-cn/web/api/urlsearchparams/index.md [hunk 4]
```diff
@@ -16,23 +16,28 @@ for (const [key, value] of mySearchParams.entries()) {
...
 
-{{availableinworkers}}
-
 ## 构造函数
 
 - {{domxref("URLSearchParams.URLSearchParams", 'URLSearchParams()')}}
   - : 返回一个 `URLSearchParams` 对象。
 
-## 方法
+## 属性
```
[the remark this answers is not in the document]
Comment (fuchunhui) [reply files/zh-cn/web/api/urlsearchparams/index.md:24]: Almost all translated as `属性` 。[Navigator](https://developer.mozilla.org/zh-CN/docs/Web/API/Navigator)

File: files/zh-cn/web/api/urlsearchparams/index.md (hunk 1 above)
Comment (yin1999) [reply files/zh-cn/web/api/urlsearchparams/index.md:129]: Yes, please!

File: files/zh-cn/web/api/urlsearchparams/index.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (yin1999) [reply files/zh-cn/web/api/urlsearchparams/index.md:24]: These legacy documents have not been formatted to the new style. Now we distinguish between _instance_ properties and _static_ properties

File: files/zh-cn/web/api/urlsearchparams/index.md [hunk 5]
```diff
@@ -105,5 +173,6 @@ searchParams3.has("query"); // true
 
 ## 参见
 
-- The {{domxref("URL")}} interface.
-- [Google Developers: Easy URL manipulation with URLSearchParams](https://developers.google.com/web/updates/2016/01/urlsearchparams?hl=en)
+- [`core-js` 中 `URLSearchParams` 的 Polyfill](https://github.com/zloirock/core-js#url-and-urlsearchparams)
+- {{domxref("URL")}} 接口
+- [谷歌开发者：使用 URLSearchParams 轻松操作 URL](https://developer.chrome.com/blog/urlsearchparams?hl=zh-cn)
```
Comment (yin1999) [inline files/zh-cn/web/api/urlsearchparams/index.md:178]:
```suggestion
- [谷歌开发者：使用 URLSearchParams 轻松操作 URL](https://developer.chrome.com/blog/urlsearchparams/)
```

**Review by yin1999 — APPROVED** [review]
Thank you @fuchunhui

**Pull request merged**